_includes/specials/taproot/en/03-p2wpkh-to-p2tr.md

```diff
@@ -0,0 +1,119 @@
+  [BIP32][] Hierarchical Deterministic (HD) code and your users don't
+  need to change their seeds.[^electrum-segwit]  However, you are
+  strongly encouraged to use a new derivation path for your P2TR public
+  keys (such as defined by [BIP86][]); if you don't do this, there's a
+  [possible attack][bip340 alt signing] that can occur if you use the
+  same keys with both ECDSA and [schnorr signatures][topic schnorr
+  signatures].
+
+- **Tweak your public key by its hash:** although technically not
+  required for single-sig, especially when all your keys are derived
+  from a randomly-chosen BIP32 seed, BIP341 [recommends][bip341 cite22]
+  having your key commit to an unspendable scripthash tree.  This is as
+  simple as using an Elliptic Curve addition operation that sums your
+  public key with the curve point of that key's hash.  Advantages of
+  complying with this recommendation are that you'll be able to use the
+  same code if you later add scriptless [multisignature][topic
+  multisignature] support or if you add support for [`tr()`
+  descriptors][].
+
+- **Create your addresses and monitor for them:** use [bech32m][topic
+  bech32] to create your addresses.  Payments will be sent to the
+  scriptPubKey `OP_1 <tweaked_pubkey>`.  You can scan for transactions
+  paying the script using whatever method you use to scan for v0 segwit
+  addresses like P2WPKH.
+
+- **Creating a spending transaction:** all the non-witness fields for
+  taproot are the same as for P2WPKH, so you don't need to worry about
+  changes to the transaction serialization.
+
+- **Create a signature message:** this is a commitment to the data from
+  the spending transaction.  Most of the data is the same as what you
+  sign for a P2WPKH transaction, but the order of the fields is
+  [changed][BIP341 sigmsg] and a few extra things are signed.
+  Implementing this is just a matter of serializing and hashing
+  various data, so writing the code should be easy.
+
+- **Sign a hash of the signature message:** there are different ways to
+  create schnorr signatures.  The best is not to "roll your own crypto"
+  but instead to use the function from a well-reviewed library you
+  trust.  But if you can't do that for some reason, [BIP340][] provides
+  an algorithm that should be straightforward to implement if you
+  already have available the primitives for making ECDSA signatures.
+  When you have your signature, put it in the witness data for your
+  input and send your spending transaction.
+
+Even before taproot activates at block {{site.trb}}, you can test your
+code using testnet, the public default [signet][topic signet], or Bitcoin Core's private
+regtest mode.  If you add taproot support to your open source wallet, we
+encourage you to add a link to the PR(s) implementing it on the [taproot
+uses][wiki taproot uses] and [bech32m adoption][wiki bech32 adoption]
+pages of the Bitcoin Wiki so other developers can learn from your code.
+
+[^electrum-segwit]:
```

Comment:
How wallets can handle different script types seems like too much good/important information to be a footnote. Perhaps we can turn this into its own post at some point in the future?

Reply:
Sure, although I don't think there's much more to say than what's here[1].  Were you thinking that's something we'd put into this taproot series, or did you have some other idea for where we'd put it?

[1] Or so I say now.  When I actually write the thing and I'm apologizing for how long it is, you can point back here and laugh at me.  :-)